perf/src/sigverify.rs

```diff
@@ -449,17 +449,29 @@ pub fn shrink_batches(batches: &mut Vec<PacketBatch>) {
     let mut last_valid_batch = 0;
     for batch_ix in 0..batches.len() {
         for packet_ix in 0..batches[batch_ix].len() {
-            if batches[batch_ix][packet_ix].meta().discard() {
+            if batches[batch_ix].get(packet_ix).unwrap().meta().discard() {
                 continue;
             }
             last_valid_batch = batch_ix.saturating_add(1);
             let mut found_spot = false;
             while valid_batch_ix < batch_ix && !found_spot {
-                while valid_packet_ix < batches[valid_batch_ix].len() {
-                    if batches[valid_batch_ix][valid_packet_ix].meta().discard() {
-                        batches[valid_batch_ix][valid_packet_ix] =
-                            batches[batch_ix][packet_ix].clone();
-                        batches[batch_ix][packet_ix].meta_mut().set_discard(true);
+                let (left, right) = batches.split_at_mut(batch_ix);
```

Comment:
think most of these changes can also go away if we implement Index on `PacketBatch`

Reply:
cannot do this easily - known "issue" w/ rust